src/app/modules/ix-table/interfaces/table-filter.interface.ts

```diff
@@ -0,0 +1,6 @@
+export interface TableFilter<T> {
+  query: string;
+  columnKeys: (keyof T)[];
+  list?: T[];
+  preprocessMap?: Partial<Record<keyof T, (value: T[keyof T]) => string>>;
+}
```

Comment:
Currently a key in `preprocessMap` is any key of T, meaning that any type is allowed on function argument as long as it exists in T.

If you change to:
```
export interface TableFilter<T> {
  query: string;
  columnKeys: (keyof T)[];
  list?: T[];
  preprocessMap?: {
    [K in keyof T]?: (value: T[K]) => string;
  };
}
```
it would require key to match function type.